sdk/core/azure_core_test/src/proxy.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//! Wrappers for the [Test Proxy](https://github.com/Azure/azure-sdk-tools/blob/main/tools/test-proxy/Azure.Sdk.Tools.TestProxy/README.md) service.
+
+use azure_core::{error::ErrorKind, Result, Url};
+use std::{
+    env, io,
+    path::Path,
+    process::{ExitStatus, Stdio},
+    sync::Arc,
+};
+use tokio::{
+    io::{AsyncBufReadExt, BufReader},
+    process::{Child, ChildStdout, Command},
+};
+use tracing::{Level, Span};
+
+// cspell:ignore aspnetcore devcert testproxy
+const KESTREL_CERT_PATH_ENV: &str = "ASPNETCORE_Kestrel__Certificates__Default__Path";
+const KESTREL_CERT_PASSWORD_ENV: &str = "ASPNETCORE_Kestrel__Certificates__Default__Password";
+const KESTREL_CERT_PASSWORD: &str = "password";
+const PROXY_MANUAL_START: &str = "PROXY_MANUAL_START";
+
+pub async fn start(
+    test_data_dir: impl AsRef<Path>,
+    options: Option<ProxyOptions>,
+) -> Result<Proxy> {
+    if env::var(PROXY_MANUAL_START).is_ok_and(|v| v.to_ascii_lowercase() == "true") {
+        return Err(azure_core::Error::message(
+            ErrorKind::Other,
+            format!("environment variable {PROXY_MANUAL_START} is 'true'; not starting test proxy"),
+        ));
```

Comment:
Should this be an error? If a user has manually started the proxy and then wants to run a test against the manually-started proxy, wouldn't this cause their test to fail, assuming there is code calling `proxy.start()?`?

Reply:
You're right. I incorrect transcribed the Go source (the intent, rather; Go isn't the problem at all) that I pulled this from.